test/scheduler/utils_test.go

```diff
@@ -462,3 +462,82 @@ func ensureProvisionalClusterDeletion(clusterName string) {
 		return err
 	}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to delete member cluster")
 }
+
+func createPickAllCRPWithPolicySnapshot(crpName string, affinity *placementv1beta1.Affinity, policySnapshotName string) {
+	policy := &placementv1beta1.PlacementPolicy{
+		PlacementType: placementv1beta1.PickAllPlacementType,
+		Affinity:      affinity,
+	}
+
+	// Create a CRP of the PickAll placement type.
+	crp := placementv1beta1.ClusterResourcePlacement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       crpName,
+			Finalizers: []string{customDeletionBlockerFinalizer},
+		},
+		Spec: placementv1beta1.ClusterResourcePlacementSpec{
+			ResourceSelectors: defaultResourceSelectors,
+			Policy:            policy,
+		},
+	}
+	Expect(hubClient.Create(ctx, &crp)).Should(Succeed(), "Failed to create CRP")
+
+	crpGeneration := crp.Generation
+
+	// Create the associated policy snapshot.
+	policySnapshot := &placementv1beta1.ClusterSchedulingPolicySnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: policySnapshotName,
+			Labels: map[string]string{
+				placementv1beta1.IsLatestSnapshotLabel: strconv.FormatBool(true),
+				placementv1beta1.CRPTrackingLabel:      crpName,
+			},
+			Annotations: map[string]string{
+				placementv1beta1.CRPGenerationAnnotation: strconv.FormatInt(crpGeneration, 10),
+			},
+		},
+		Spec: placementv1beta1.SchedulingPolicySnapshotSpec{
+			Policy:     policy,
+			PolicyHash: []byte(policyHash),
+		},
+	}
+	Expect(hubClient.Create(ctx, policySnapshot)).Should(Succeed(), "Failed to create policy snapshot")
+}
+
+func updatePickAllCRPWithNewAffinity(crpName string, affinity *placementv1beta1.Affinity, oldPolicySnapshotName, newPolicySnapshotName string) {
+	// Update the CRP.
+	crp := &placementv1beta1.ClusterResourcePlacement{}
+	Expect(hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp)).To(Succeed(), "Failed to get CRP")
+
+	policy := crp.Spec.Policy.DeepCopy()
+	policy.Affinity = affinity
+	crp.Spec.Policy = policy
```

Comment:
```suggestion
	crp.Spec.Policy.Affinity = affinity
```

Reply:
The policy struct is also used in a different place, hence the deepcopy.